### ISSUE TITLE
Link tooltip detection vertically misaligned

### ISSUE THREAD

**Issue by tridian-tn** [zim-desktop-wiki/zim-desktop-wiki#2484]
**Describe the bug**
The mouse hover detection area for link tooltips don't align with the links themselves.

Best described with the attached gif: when you have links on a page, I'm finding that the popup tooltip doesn't correspond with the link that the mouse is over. It looks like the detection is "off" vertically.

It doesn't affect the functioning of the links: they still work correctly when you click them. There's no vertical offset there.

**To Reproduce**
1. Put some links on a page
2. Hover the mouse over those links and wait for the tooltips to come up

**Expected behavior**
The correct tooltip should show when the mouse is over the link, not when it's some way below it.

**Screenshots**
![Zim](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/10810780/b79a071d-6f4c-41e4-8ed9-5a8489038163)

**Environment (please complete the following information):**
 - Version 0.75.2 release
 - OS: Windows 11
 - Language: English / GB

**Additional context**
Tested with my custom style.conf, and with the default style.conf

Nothing logged in the debug log when this happens.

**Comment by introt** [comment]
Can you replicate this with gtk3-demo? Might be Windows 11 spesific

**Comment by tridian-tn** [comment]
Seems to be ok, if this is what you were thinking of. No tooltip, but the detection of the link is ok when the mouse hovers over it
![Zim](https://github.com/zim-desktop-wiki/zim-desktop-wiki/assets/10810780/b92ae06e-b4ba-4bf7-868c-e922cd89b950)

**Comment by introt** [comment]
You got it, thanks. From your first video it seems like the problem isn't limited to the tool tips - the correct cursor isn't displayed while hovering either. If you're willing to dig around, enabling CSS (toggling "Show controls in the window decoration" under the Tool Bar plugin) and/or disabling the Bookmarks Bar plugin might have an effect.

**Comment by tridian-tn** [comment]
Thanks - I've tried those, and turned off the toolbar plugin, and the pathbar as well (just in case!). No effect - the cursor and tooltip detection are still askew.

**Comment by introt** [comment]
The gtk3-demo might use a different version of the toolkit than Zim's Windows build, so the next steps would be to see if this issue is present on other Windows 11 systems (a clean user profile might suffice) and if so, building the demo and Zim with the same toolchain to rule out it being an upstream issue.

**Comment by tridian-tn** [comment]
Ok thanks. I've tried it with the msys2 gtk3-demo, under the same msys2 environment I used to run Zim from source when I tested to see if the result was the same. If I need to do more than that then time may be an issue!

**Comment by introt** [comment]
That does point very strongly to it being Zim-specific then. Since I can't reproduce it on Linux, I'm going to tag this issue Windows so someone running it might be able to confirm and fix it.

**Comment by bruvellu** [comment]
I had this issue on Linux at some point. But I can't remember if I figured it out or if it just went away. Have you tried changing the zoom level (font size) to see if the offset changes? Or, if you are on a HiDPI screen, try lowering to a regular resolution.

**Comment by tridian-tn** [comment]
> I had this issue on Linux at some point. But I can't remember if I figured it out or if it just went away. Have you tried changing the zoom level (font size) to see if the offset changes? Or, if you are on a HiDPI screen, try lowering to a regular resolution.

It's a standard DPI screen, and I've just tried changing the zoom level in and out, and the effect is the same, unfortunately. I tried a few prior versions - 0.68 definitely works properly, and I went back through the git log to 0.74 which still has the problem. I was getting errors when I went back further, so I'm guessing I'd need to wipe my local config first

**Comment by roland-ruedenauer** [comment]
I'm using a custom style too with current Zim and can confirm the problem also exists on Linux (Manjaro).

**Comment by introt** [comment]
@roland-ruedenauer could you try to reproduce this with the flatpak version? Perhaps we could narrow this down to specific library versions or style options. Do also include whether you're using X11 or Wayland.

**Comment by roland-ruedenauer** [comment]
I will have a closer look in the next days. And I use Zim in a Wayland desktop session.

**Comment by roland-ruedenauer** [comment]
As promised, I had a look what caused the problem on my installation and found the culprit (at least in my case):
a plugin I did installed long time ago modified the border-width property of the main TextView widget.
This led to zim miscalculate mouse pointer coordinates in the pageview package.

The fix for the problem is rather simple: change [line 4242 at zim/gui/pageview/init.py](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/0.75.2/zim/gui/pageview/__init__.py#L4242) from
```
coords = self.window_to_buffer_coords(Gtk.TextWindowType.WIDGET, x, y)
```
into
```
coords = self.window_to_buffer_coords(Gtk.TextWindowType.TEXT, x, y)
```

You may reproduce the issue easily using the Gtk Inspector: start zim using `GTK_DEBUG=interactive zim` then change the `border-width` property of object zim-pageview object and see the cursor display being displaced by the textview border-width.

Btw, the plugin I was using to style the TextView is available here: https://omacronides.com/projets/zim-plugin-textview-prefs/

Hope this helps.

**Comment by introt** [comment]
Thank you Roland! Could you submit the fix as a PR? Feel free to use the in-browser editor.

**Comment by tridian-tn** [comment]
@introt @roland-ruedenauer I can confirm that Roland's PR also fixes the issue for me on Windows. And the Textview plugin turned out to be the culprit here too. Thanks Roland - very much appreciated

**Comment by tridian-tn** [comment]
As an unrelated aside, having disabled that plugin I can see that editor font rendering on Windows is improved: it's visibly better defined when the plugin is disabled, for whatever reason. That was a source of some irritation for me in recent weeks so this PR has indirectly helped with that too!

**Comment by introt** [comment]
Fixed by  #2484
